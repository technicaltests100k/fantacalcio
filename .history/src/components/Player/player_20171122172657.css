.player {
  width: 4vh;
  height: 4vh;
  background-color: red;
  margin: 2vmax;
}
