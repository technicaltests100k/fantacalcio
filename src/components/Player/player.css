.player {
  width: 4vh;
  height: 4vh;
  margin: 2vmax;
}
