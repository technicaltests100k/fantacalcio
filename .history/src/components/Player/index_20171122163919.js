import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './player.css';

/**
 *	Player React Type component.
 */
export class Player extends Component {
  /**
   *	Props implementation.
   */
  static propTypes = {};
  /**
   * Implements defaultProps().
   */
  static defaultProps = {};
  /**
   *	Player contructor.
   * @param {object} props to constructor.
   *	@constructor
   */
  constructor(props) {
    super(props);
  }
  /**
   * Render
   * @return {ReactElement} markup
   */
  render() {
    return <div className={'player'} />;
  }
}

export default Player;
