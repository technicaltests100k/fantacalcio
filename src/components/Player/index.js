import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './player.css';
import tshirtImg from '../../Assets/player-shirt.png';

/**
 *	Player React Type component.
 */
export class Player extends PureComponent {
  /**
   *	Props implementation.
   */
  static propTypes = {
    player: PropTypes.object,
    tshirt: PropTypes.bool
  };
  /**
   * Implements defaultProps().
   */
  static defaultProps = {
    tshirt: false
  };

  /**
   * Render
   * @return {ReactElement} markup
   */
  render() {
    const { tshirt, player } = this.props;
    const { name, position, type } = player;

    return tshirt ? (
      <div className={'player'}>
        <img src={tshirtImg} alt="player tshirt" />
        <p>{name}</p>
      </div>
    ) : (
      <li className={'playerList'}>
        <p>{`${name} - ${position} - ${type}`}</p>
      </li>
    );
  }
}

export default Player;
