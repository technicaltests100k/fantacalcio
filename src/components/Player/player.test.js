import React from 'react';
import Player from './';
/**
 * Test for <Player /> component.
 * @test {Player}
 */
const props = {
  player: {
    name: 'Estebian Cambiasso',
    position: 'Midfielder',
    type: 'CM'
  },
  tshirt: true
};
describe('<Player />', () => {
  it('should render Player with its tshirt img', () => {
    const wrapper = render(<Player {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Player as an li element', () => {
    const wrapper = render(<Player {...props} tshirt={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
