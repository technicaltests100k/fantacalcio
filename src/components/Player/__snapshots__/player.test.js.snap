// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> should render Player as an li element 1`] = `
<li
  class="playerList"
>
  <p>
    Estebian Cambiasso - Midfielder - CM
  </p>
</li>
`;

exports[`<Player /> should render Player with its tshirt img 1`] = `
<div
  class="player"
>
  <img
    alt="player tshirt"
    src="player-shirt.png"
  >
    
  </img>
  <p>
    Estebian Cambiasso
  </p>
</div>
`;
